src/mozanalysis/metrics/__init__.py

```diff
@@ -90,6 +98,24 @@ def get_sanity_metrics(self, experiment_slug):
                 ),
             ]
 
+        elif self.experiments_column_type == 'map_struct':
+            return [
+                Metric(
+                    name=self.name + '_has_contradictory_branch',
+                    data_source=self,
+                    select_expr=agg_any("""`moz-fx-data-shared-prod.udf.get_key`(
+                ds.experiments, '{experiment_slug}'
+            ).branch != e.branch"""),
+                ),
+                Metric(
+                    name=self.name + '_has_non_enrolled_data',
+                    data_source=self,
+                    select_expr=agg_any("""`moz-fx-data-shared-prod.udf.get_key`(
+                ds.experiments, '{experiment_slug}'
+            ).branch IS NULL""".format(experiment_slug=experiment_slug))
```

Comment:
This indentation is a little hard to follow but I assume it does something nice looking on the output side